src/iris/api.py

```diff
@@ -5430,10 +5438,82 @@ def on_get(self, req, resp):
         resp.body = ujson.dumps(payload)
 
 
+class InternalIncident():
+    allow_read_no_auth = False
+
+    def __init__(self, config):
+        self.allowlisted_apps = config.get('allowlisted_internal_apps', [])
+
+    def on_post(self, req, resp, incident_id):
+        if req.context.get('app', {}).get('name') not in self.allowlisted_apps:
+            raise HTTPUnauthorized('Authentication failure', 'this endpoint is only available for internal use by allowlisted apps', [])
+
+        body = ujson.loads(req.context['body'])
+
+        if 'current_step' not in body or 'active' not in body:
+            raise HTTPBadRequest('Missing required fields "current_step" or "active" in body')
+
+        if not isinstance(body['current_step'], int):
+            raise HTTPBadRequest('"step" field must be an integer"')
+
+        connection = db.engine.raw_connection()
+        cursor = connection.cursor(db.dict_cursor)
+
+        cursor.execute(''' SELECT EXISTS (SELECT `id` from `incident` where `id` = %s)''', incident_id)
+        if cursor.fetchone() is None:
```

Comment:
Because this is doing a ` select exists()` won't the return value always be a single-element array containing a 1 or 0 as opposed to something that could be none?